src/handleAction.js

```diff
@@ -1,28 +1,22 @@
-function isFunction(val) {
-  return typeof val === 'function';
-}
+import isFunction from 'lodash.isfunction';
+import identity from 'lodash.identity';
+import isNil from 'lodash.isnil';
+import isSymbol from 'lodash.issymbol';
 
 export default function handleAction(type, reducers, defaultState) {
-  const typeValue = isFunction(type)
-    ? type.toString()
-    : type;
-
-  return (state = defaultState, action) => {
-    // If action type does not match, return previous state
-    if (action.type !== typeValue) return state;
+  const typeValue = isSymbol(type)
+    ? type
+    : type.toString();
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
+  const [nextReducer, throwReducer] = isFunction(reducers)
+    ? [reducers, reducers]
+    : [reducers.next, reducers.throw].map(reducer => (isNil(reducer) ? identity : reducer));
```

Comment:
Alright, I've gone through that again. I understand your idea know behind this. Leave it like that ;)